build/storage/scripts/disk_infrastructure.py

```diff
@@ -0,0 +1,207 @@
+import base64
+import importlib
+import logging
+import re
+import socket
+import sys
+import time
+import uuid
+
+from scripts import socket_functions
+
+sys.path.append("/usr/libexec/spdk/scripts")
+sma_client = importlib.import_module("sma-client")
+import rpc
+
+logging.root.setLevel(logging.CRITICAL)
+
+
+def get_number_of_virtio_blk(sock: str) -> int:
+    cmd = 'lsblk --output "NAME"'
+    out = socket_functions.send_command_over_unix_socket(
+        sock=sock, cmd=cmd, wait_for_secs=1
+    )
+    number_of_virtio_blk_devices = len(re.findall("vd", out))
+    return number_of_virtio_blk_devices
+
+
+def is_virtio_blk_attached(sock: str) -> int:
+    if get_number_of_virtio_blk(sock) == 0:
+        logging.error("virtio-blk is not found")
+        return 1
+    logging.info("virtio-blk is found")
+    return 0
+
+
+def is_virtio_blk_not_attached(sock: str) -> int:
+    if is_virtio_blk_attached(sock):
+        return 0
+    return 1
+
+
+def check_number_of_virtio_blk_devices(
```

Comment:
rename to 'verify_'? check sounds more like 'get'

Reply:
Sure, right now it's called the same as in the main ipdk-io repo. I will change it.